### PR TITLE
Add Prometheus metrics and switch to vanilla libp2p

### PR DESCRIPTION
**Pull request by folex**
Also, update `air-interpreter-wasm = "=0.20.2"`

### CODE REVIEW

File: aquamarine/src/particle_functions.rs [hunk 1]
```diff
@@ -36,11 +36,26 @@ use particle_execution::{
 pub type Function =
     Box<dyn FnMut(Args, ParticleParams) -> ParticleFunctionOutput<'static> + 'static + Send + Sync>;
 
+#[derive(Clone, Debug)]
+pub struct SingleCallStat {
+    pub run_time: Option<Duration>,
+    pub success: bool,
+    pub builtin: bool,
```
Comment (justprosh) [inline aquamarine/src/particle_functions.rs:43]: what does it mean? Maybe add some doc comment

File: connection-pool/src/api.rs [hunk 2]
```diff
@@ -70,14 +71,18 @@ pub enum Command {
     },
 }
 
-pub type SwarmEventType = generate_swarm_event_type!(ConnectionPoolInlet);
+// pub type SwarmEventType = generate_swarm_event_type!(ConnectionPoolInlet);
```
Comment (justprosh) [inline connection-pool/src/api.rs:74]: dead code?

File: crates/connected-client/src/client.rs [hunk 3]
```diff
@@ -157,7 +159,7 @@ impl Client {
                     to_relay = relay_inlet.next() => {
                         if let Some(cmd) = to_relay {
                             // Send to node
```
Comment (justprosh) [inline crates/connected-client/src/client.rs:161]:
```suggestion
```
comment just duplicates function name

**Review by justprosh — APPROVED** [review]
don't forget to change name in config in [node-distro](https://github.com/fluencelabs/node-distro/blob/main/fluence/Config.default.toml#L50)

File: aquamarine/src/particle_functions.rs (hunk 1 above)
Comment (folex) [reply aquamarine/src/particle_functions.rs:43]:
```rust
/// Performance statistics about executed function call
pub struct SingleCallStat {
    /// If execution happened, then how much time it took
    pub run_time: Option<Duration>,
    pub success: bool,
    /// Whether function call was to builtin functions (like op noop) or to services
    pub kind: FunctionKind,
}
```

File: particle-node/src/node.rs [hunk 4]
```diff
@@ -113,20 +115,31 @@ impl<RT: AquaRuntime> Node<RT> {
...
-        let metrics_registry = if config.prometheus_config.prometheus_enabled {
-            Some(Registry::new())
+        let mut metrics_registry = if config.metrics_config.metrics_enabled {
+            Some(Registry::default())
         } else {
             None
         };
-        let network_config =
-            NetworkConfig::new(metrics_registry.clone(), key_pair, &config, node_version);
+        let libp2p_metrics = metrics_registry.as_mut().map(Metrics::new);
```
Comment (kmd-fl) [inline particle-node/src/node.rs:123]: libp2p_metrics are created here

File: particle-node/src/node.rs [hunk 5]
```diff
@@ -288,25 +304,34 @@ impl<RT: AquaRuntime> Node<RT> {
...
+        let mut registry = self.registry;
         let metrics_listen_addr = self.metrics_listen_addr;
         let local_peer_id = self.local_peer_id;
         let builtins_management_peer_id = self.builtins_management_peer_id;
 
         task::spawn(async move {
-            let mut metrics = if let Some(registry) = registry {
-                start_metrics_endpoint(registry, metrics_listen_addr)
+            let (metrics_fut, libp2p_metrics) = if let Some(mut registry) = registry {
+                let libp2p_metrics = Metrics::new(&mut registry);
```
Comment (kmd-fl) [inline particle-node/src/node.rs:314]: and here

Is it okay? It may be connected to doubling libp2p metrics

**Pull request merged**